### ISSUE TITLE
LTHR String

### ISSUE THREAD

**Issue by etoumey** [jtoumey/tss_estimator#4]
don't input string or char for LTHR or else infinite loop. Trying to fix now.

**Comment by jtoumey** [comment]
Cool. haha. push your branch when you have sec, I think it's `git push origin "branch_name"`

**Comment by jtoumey** [comment]
I'm a fred programmer. I think `scanf` if not what we should use...

http://stackoverflow.com/questions/18210406/check-if-input-is-float-else-stop

**Comment by etoumey** [comment]
Yeah I read that too. Here are my thoughts. Forget inputting the file name, gives rise to too many problems. We should just pass the file directly in. I'm pretty sure that way, any time you say "scanf" it snags whatever is next in the file.

So in the executable it would be like 

<pre><code> gcc get_input.c calc_hr_zones.c main.c -lm
./2015-11-10-341232131.gpx > a.out
</code></pre>

It's something like that, but we can use bash to use system date, assuming u r calcing on the day u did the activity

**Comment by jtoumey** [comment]
Good call. We are getting sidetracked on this and it really doesn't matter that much. 

Ok so when you say _we can use bash to use system date, assuming u r calcing on the day u did the activity_, do you mean we write a bash script that, when run, uses system date to create the *_.gpx_ file name for input? so you would run a bash script and it would automatically do all the stuff ? (like you said, assuming you run the same day as activity)

**Comment by etoumey** [comment]
Give me a sec and i'll push a commit

**Comment by etoumey** [comment]
DANG IT im on windows. 

give me a few seconds and i'll push a commit lol

**Comment by jtoumey** [comment]
LOL :dancer:

**Comment by etoumey** [comment]
okay, i can actually write this out later, i dont want to switch back to linux cuz i dont have a good MATLAB over there, but here's the concept

<pre><code>#!/bin/bash
FILE=$(date +"%Y-%m-%d.gpx")
gcc ~/Documents/tss_estimator/src/get_input.c ~/Documents/tss_estimator/src/calc_hr_zone.c ~/Documents/tss_estimator/src/main.c
./~/Documents/tss_estimator/src/a.out < /Path/to/GARMIN/FILE 
</code></pre>

**Comment by jtoumey** [comment]
OK, dope, that's great. That will save a lot of time :) 

I was thinking, at this point for testing we can hard-code file names, LTHRs, etc. if we need to, just until we get everything working.

**Comment by etoumey** [comment]
That's fair enough. Do you still want to include bash script stuff? 

also, you could put a .gpx in the repo and hardcode that in the bash. The "make" -> cd bin -> ./main was getting a little exhausting, this way you can just type "tss" or something from ANY directory and get the results of the prog

**Comment by jtoumey** [comment]
Including the bash script stuff sounds good. Tonight I can add the test *_.gpx_ you sent me. 

If you could add your run script, that would be coo ;)

**Comment by etoumey** [comment]
will probs do 2nite. maybe this afternoon
